app/member-reminders/member-reminders-public/src/main/kotlin/com/hedvig/android/memberreminders/GetMemberRemindersUseCase.kt

```diff
@@ -20,79 +24,89 @@ internal class GetMemberRemindersUseCaseImpl(
         emit(upcomingRenewals)
       },
       flow {
-        val coInsuredInfo = getNeedsCoInsuredInfoRemindersUseCase.invoke().getOrNull()?.let {
-          MemberReminder.CoInsuredInfo(it)
-        }
+        val coInsuredInfo = getNeedsCoInsuredInfoRemindersUseCase.invoke().getOrNull()
         emit(coInsuredInfo)
       },
     ) {
         enableNotifications: MemberReminder.EnableNotifications?,
         connectPayment: MemberReminder.ConnectPayment?,
-        upcomingRenewalReminders: MemberReminder.UpcomingRenewals?,
-        coInsuredInfo: MemberReminder.CoInsuredInfo?,
+        upcomingRenewalReminders: NonEmptyList<MemberReminder.UpcomingRenewal>?,
+        coInsuredInfo: NonEmptyList<MemberReminder.CoInsuredInfo>?,
       ->
       MemberReminders(
         connectPayment = connectPayment,
         upcomingRenewals = upcomingRenewalReminders,
         enableNotifications = enableNotifications,
-        coInsuredInfo = coInsuredInfo,
+        coInsuredInfo = listOfNotNull(coInsuredInfo?.first()),
       )
     }
   }
 }
 
 data class MemberReminders(
   val connectPayment: MemberReminder.ConnectPayment? = null,
-  val upcomingRenewals: MemberReminder.UpcomingRenewals? = null,
+  val upcomingRenewals: List<MemberReminder.UpcomingRenewal>? = null,
   val enableNotifications: MemberReminder.EnableNotifications? = null,
-  val coInsuredInfo: MemberReminder.CoInsuredInfo? = null,
+  val coInsuredInfo: List<MemberReminder.CoInsuredInfo>? = null,
 ) {
   /**
    * In some cases a reminder may be present but may not be applicable in our current app state.
    *
    * If [alreadyHasNotificationPermission] is true, then the notification permission reminder should not be shown.
    */
-  fun onlyApplicableReminders(alreadyHasNotificationPermission: Boolean): ApplicableMemberReminders {
-    return ApplicableMemberReminders(
-      connectPayment,
-      upcomingRenewals,
-      enableNotifications.takeIf { !alreadyHasNotificationPermission },
-      coInsuredInfo,
-    )
+  fun onlyApplicableReminders(alreadyHasNotificationPermission: Boolean): ImmutableList<MemberReminder> {
+    return buildList {
+      connectPayment?.let {
+        add(connectPayment)
+      }
+      coInsuredInfo?.let {
+        addAll(coInsuredInfo)
+      }
+      if (alreadyHasNotificationPermission) {
+        enableNotifications?.let {
+          add(enableNotifications)
+        }
+      }
+      upcomingRenewals?.let {
+        addAll(it)
+      }
+    }.toImmutableList()
   }
 }
 
-data class ApplicableMemberReminders(
-  val connectPayment: MemberReminder.ConnectPayment? = null,
-  val upcomingRenewals: MemberReminder.UpcomingRenewals? = null,
-  val enableNotifications: MemberReminder.EnableNotifications? = null,
-  val coInsuredInfo: MemberReminder.CoInsuredInfo? = null,
-) {
-  val hasAnyReminders: Boolean
-    get() = connectPayment != null || upcomingRenewals != null || enableNotifications != null || coInsuredInfo != null
-}
```

Comment:
This class existed to model the fact that it's not a possible state to have a list of `[MemberReminder.ConnectPayment, MemberReminder.UpcomingRenewals, MemberReminder.ConnectPayment]` for example.
With the new Api introduced into this PR, we're making it possible for us to pass a list with 2 ConnectPayment objects, which should never be something that should be possible.
Do we feel the need for this API to change like this to stop modeling this limitation in the type itself?

Reply:
I see, thats a valid point. `GetMemberRemindersUseCase` will still return 
```
data class MemberReminders(
  val connectPayment: MemberReminder.ConnectPayment? = null,
  val upcomingRenewals: List<MemberReminder.UpcomingRenewal>? = null,
  val enableNotifications: MemberReminder.EnableNotifications? = null,
  val coInsuredInfo: List<MemberReminder.CoInsuredInfo>? = null,
)
```
so I think that model is still clear. The view can then take a list of `MemberReminder`, and dont care about the ordering or det underlying construction of that list. It made it quite a bit easier to work with in the view.